libs/config/src/analyze/lints/c14_unused_external.rs

```diff
@@ -0,0 +1,243 @@
+use std::{cell::RefCell, rc::Rc, sync::Arc};
+
+use hemtt_common::config::{LintConfig, ProjectConfig};
+use hemtt_workspace::{
+    lint::{AnyLintRunner, Lint, LintRunner},
+    reporting::{Code, Codes, Diagnostic, Processed, Severity},
+};
+use indexmap::IndexMap;
+
+use crate::{analyze::LintData, Class, Config, Property};
+
+crate::analyze::lint!(LintC14UnusedExternal);
+
+impl Lint<LintData> for LintC14UnusedExternal {
+    fn ident(&self) -> &'static str {
+        "unused_external"
+    }
+    fn sort(&self) -> u32 {
+        140
+    }
+    fn description(&self) -> &'static str {
+        "Reports on external classes that are never used"
+    }
+    fn documentation(&self) -> &'static str {
+        "### Example
+
+**Incorrect**
+```hpp
+class x;
+```
+
+**Correct**
+```hpp
+```
+"
+    }
+    fn default_config(&self) -> LintConfig {
+        LintConfig::help().with_enabled(hemtt_common::config::LintEnabled::Pedantic)
```

Comment:
Is there a good reason for pedantic?

Would this be prone to false positives or genuine reasons to exist?

Reply:
They don't really create any problems and I don't want to force people to change good working configs
vehicle turret configs especially may require many lines changed

but I still think it can be a useful optional tool to show configs that may be inheriting from the wrong level
or if people really want to cleanup as much as possible

Reply:
Perhaps a thing where if there are more than x Disha from this lint, we suppress them by default and tell the user to either turn it off or explicitly turn it on?